File: api/src/RadioactivityApp.BfsClient/Models/Feature.cs
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;

namespace RadioactivityApp.BfsClient.Models;

public sealed record Feature
{
    [Required, JsonInclude, JsonPropertyName("id")]
    public string? Id { get; init; }

    [Required, JsonInclude, JsonPropertyName("geometry")]
    public GeometryPoint? Geometry { get; init; }

    [Required, JsonInclude, JsonPropertyName("geometryName")]
    public string? GeometryName { get; init; }

    [Required, JsonInclude, JsonPropertyName("properties")]
    public FeatureProperties? Properties { get; init; }
}

File: app/lib/api/models/feature.dart
import 'package:meta/meta.dart';

import 'geometry.dart';
import 'properties.dart';

@sealed
@immutable
class Feature {
  final String id;
  final GeometryPoint geometry;
  final String geometryName;
  final FeatureProperties properties;

  Feature(this.id, this.geometry, this.geometryName, this.properties);

  Feature.fromJson(Map<String, dynamic> json)
      : id = json['id'] as String,
        geometry = GeometryPoint.fromJson(json['geometry']),
        geometryName = json['geometry_name'] as String,
        properties = FeatureProperties.fromJson(json['properties']);
}
